nemo_aligner/models/nlp/gpt/megatron_gpt_ppo_actor.py

```diff
@@ -200,14 +201,35 @@ def finish_training(self):
         """no need to offload adam states here
         """
 
+    def _apply_sampling_params(self, logits, context_lengths):
+        samparams = self._sampling_params
+        # apply the sampling params to the logits - focusing only on the generated tokens.
+        context_length = context_lengths.min().item()
+        resp_logits = logits[:, context_length - 1 :].contiguous()
+        if not samparams.get("use_greedy", False):  # if use_greedy is True, use the logits as is
```

Comment:
Minor suggestion: move this up two lines and write it
```python
if samparams.get("use_greedy", False):
    return logits
```
which will avoid a couple of useless ops & extra indent

Reply:
Also: I think we should add a check to skip scaling if temp == 1, and top_p / top_k if they are equal to 1.0 / 1. This way we don't mess with logits for no good reason.